src/nvim/fileio.c

```diff
@@ -4447,11 +4448,95 @@ bool vim_fgets(char_u *buf, int size, FILE *fp) FUNC_ATTR_NONNULL_ALL
   return eof == NULL;
 }
 
-/*
- * os_rename() only works if both files are on the same file system, this
- * function will (attempts to?) copy the file across if rename fails -- webb
- * Return -1 for failure, 0 for success.
- */
+/// Read 2 bytes from "fd" and turn them into an int, MSB first.
+int get2c(FILE *fd)
+{
+  int n;
+
+  n = getc(fd);
+  n = (n << 8) + getc(fd);
+  return n;
+}
+
+/// Read 3 bytes from "fd" and turn them into an int, MSB first.
+int get3c(FILE *fd)
+{
+  int n;
+
+  n = getc(fd);
+  n = (n << 8) + getc(fd);
+  n = (n << 8) + getc(fd);
+  return n;
+}
+
+/// Read 4 bytes from "fd" and turn them into an int, MSB first.
+int get4c(FILE *fd)
+{
+  // Use unsigned rather than int otherwise result is undefined
+  // when left-shift sets the MSB.
+  unsigned n;
+
+  n = (unsigned)getc(fd);
+  n = (n << 8) + (unsigned)getc(fd);
+  n = (n << 8) + (unsigned)getc(fd);
+  n = (n << 8) + (unsigned)getc(fd);
+  return (int)n;
+}
+
+/// Read 8 bytes from `fd` and turn them into a time_t, MSB first.
+time_t get8ctime(FILE *fd)
+{
+  time_t n = 0;
+  int i;
+
+  for (i = 0; i < 8; i++) {
+    n = (n << 8) + getc(fd);
+  }
+  return n;
+}
+
+/// Reads a string of length "cnt" from "fd" into allocated memory.
+/// @return pointer to the string or NULL when unable to read that many bytes.
+char *read_string(FILE *fd, size_t cnt)
+{
+  uint8_t *str = xmallocz(cnt);
```

Comment:
I would not suggest to use `uint8_t` for *string* values. Though specifically in this case I do not see why it is not `char *`, but if you meant “string with unsigned characters” it is `char_u *` (and should be avoided if using `char` is possible). `uint8_t *` looks like “array of numbers”, not “string with unsigned characters”.